geode-assembly/src/acceptanceTest/java/org/apache/geode/cache/persistence/MissingDiskStoreAcceptanceTest.java

```diff
@@ -0,0 +1,153 @@
+import static org.apache.geode.test.awaitility.GeodeAwaitility.await;
+import static org.apache.geode.test.awaitility.GeodeAwaitility.getTimeout;
+
+import java.nio.file.Path;
+import java.util.concurrent.Future;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.internal.InternalClientCache;
+import org.apache.geode.test.assertj.LogFileAssert;
+import org.apache.geode.test.junit.rules.ExecutorServiceRule;
+import org.apache.geode.test.junit.rules.gfsh.GfshRule;
+
+public class MissingDiskStoreAcceptanceTest {
+
+  private static final String SERVER_1_NAME = "server1";
+  private static final String SERVER_2_NAME = "server2";
+  private static final String LOCATOR_NAME = "locator";
+  private static final String REGION_NAME = "myRegion";
+
+  private InternalClientCache clientCache;
+
+  private Path locatorFolder;
+  private Path server1Folder;
+  private Path server2Folder;
+
+  private int locatorPort;
+
+  private String startServer1Command;
+  private String startServer2Command;
+
+  @Rule
+  public ExecutorServiceRule executorServiceRule = new ExecutorServiceRule();
+
+  @Rule
+  public GfshRule gfshRule = new GfshRule();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void setUp() throws Exception {
+    locatorFolder = temporaryFolder.newFolder(LOCATOR_NAME).toPath().toAbsolutePath();
+    server1Folder = temporaryFolder.newFolder(SERVER_1_NAME).toPath().toAbsolutePath();
+    server2Folder = temporaryFolder.newFolder(SERVER_2_NAME).toPath().toAbsolutePath();
+
+    int[] ports = getRandomAvailableTCPPorts(3);
+    locatorPort = ports[0];
+    int server1Port = ports[1];
+    int server2Port = ports[2];
+
+    String startLocatorCommand = String.join(" ",
+        "start locator",
+        "--name=" + LOCATOR_NAME,
+        "--dir=" + locatorFolder,
+        "--port=" + locatorPort,
+        "--locators=localhost[" + locatorPort + "]");
+
+    startServer1Command = String.join(" ",
+        "start server",
+        "--name=" + SERVER_1_NAME,
+        "--dir=" + server1Folder,
+        "--locators=localhost[" + locatorPort + "]",
+        "--server-port=" + server1Port);
+
+    startServer2Command = String.join(" ",
+        "start server",
+        "--name=" + SERVER_2_NAME,
+        "--dir=" + server2Folder,
+        "--locators=localhost[" + locatorPort + "]",
+        "--server-port=" + server2Port);
+
+    String createRegionCommand = String.join(" ",
+        "create region",
+        "--name=" + REGION_NAME,
+        "--type=REPLICATE_PERSISTENT");
+
+    gfshRule.execute(startLocatorCommand, startServer1Command, startServer2Command,
+        createRegionCommand);
+
+    clientCache = (InternalClientCache) new ClientCacheFactory()
```

Comment:
Why is this casting to InternalClientCache?

Reply:
Removed. It just uses ClientCache so I'm not sure why the field was InternalClientCache.